lambda/handler.py

```diff
@@ -0,0 +1,71 @@
+import json
+import mimetypes
+import os
+import os.path
+import tempfile
+import time
+
+import boto3
+
+import dumb_pypi.main
+
+
+def _load_config():
+    with open(os.path.join(os.path.dirname(__file__), 'config.json')) as f:
+        return json.load(f)
+
+
+def _list_bucket(bucket):
+    s3 = boto3.client('s3')
+    paginator = s3.get_paginator('list_objects_v2')
+    for page in paginator.paginate(Bucket=bucket):
+        yield from (
+            json.dumps(
+                {
+                    'filename': package['Key'],
+                    'upload_timestamp': time.mktime(package['LastModified'].timetuple()),
+                },
+                sort_keys=True,
+            )
+            for package in page.get('Contents', ())
+        )
+
+
+def _sync_bucket(localdir, bucket_name):
```

Comment:
I was looking at how `aws s3 sync` works and it basically does a `(mtime, size)` comparison against the remote objects -- going to try and implement something like that as well :)